app/src/main/java/org/fossasia/openevent/app/data/sponsor/SponsorDelegate.java

```diff
@@ -0,0 +1,18 @@
+package org.fossasia.openevent.app.data.sponsor;
+
+import android.view.View;
+
+import org.fossasia.openevent.app.common.model.HeaderProvider;
+import org.fossasia.openevent.app.core.sponsor.list.viewholder.SponsorsViewHolder;
+
+import java.util.List;
+
+public interface SponsorDelegate extends Comparable<Sponsor>, HeaderProvider {
+
+    long getIdentifier();
+    int getType();
+    int getLayoutRes();
+    SponsorsViewHolder getViewHolder(View view);
+    void bindView(SponsorsViewHolder holder, List<Object> list);
+    void unbindView(SponsorsViewHolder holder);
```

Comment:
Don't add these many methods in it. Some don't even make sense to be called on sponsor object. These do not belong in the data layer of the app